crates/bevy_ecs/src/component.rs

```diff
@@ -440,7 +456,8 @@ impl Components {
     ) -> usize {
         let index = components.len();
         let info = ComponentInfo::new(ComponentId(index), descriptor);
-        if info.descriptor.storage_type == StorageType::SparseSet {
+        // Resources are never stored in sparse sets.
+        if info.descriptor.storage_type == StorageType::SparseSet && !info.descriptor.is_resource {
```

Comment:
this may change in the future (see [this perf comment](https://github.com/bevyengine/bevy/blob/1c5c94715cb17cda5ae209eef12a938501de90b5/crates/bevy_ecs/src/component.rs#L365-L366)), and why add the test anyway?

Reply:
This is an artefact of some changes I made to `ComponentDescriptor::new_with_layout` that I ended up reverting.